### PR TITLE
feat: bigframes.bigquery.json_extract

### PR DESCRIPTION
**Pull request by chelsea-lin**
BEGIN_COMMIT_OVERRIDE
feat: Implement `bigframes.bigquery.json_extract` (#868)
END_COMMIT_OVERRIDE

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue: Internal issue 346571659
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes internal bug: 346571659 🦕

### CODE REVIEW

File: tests/system/small/bigquery/test_json.py [hunk 1]
```diff
@@ -110,3 +110,30 @@ def test_json_set_w_invalid_value_type():
...
+
+
+def test_json_extract_from_string():
+    s = bpd.Series(['{"a": {"b": [1, 2]}}', '{"a": {"c": 1}}', '{"a": {"b": 0}}'])
+    actual = bbq.json_extract(s, "$.a.b")
+    expected = _get_series_from_json(["[1,2]", None, "0"])
+    pd.testing.assert_series_equal(
+        actual.to_pandas(),
+        expected.to_pandas(),
+        check_names=False,
```
Comment (shobsi) [inline tests/system/small/bigquery/test_json.py:133]: why do we need this here but not in `test_json_extract_from_json`?

File: bigframes/operations/__init__.py [hunk 2]
```diff
@@ -602,6 +602,22 @@ def output_type(self, *input_types):
         return dtypes.STRING_DTYPE
 
 
+## JSON Ops
+@dataclasses.dataclass(frozen=True)
+class JSONExtract(UnaryOp):
+    name: typing.ClassVar[str] = "json_extract"
+    json_path: str
+
+    def output_type(self, *input_types):
```
Comment (shobsi) [inline bigframes/operations/__init__.py:611]: what would be a good example of multiple input_types? Can they be different?

File: bigframes/bigquery/__init__.py [hunk 3]
```diff
@@ -208,6 +208,41 @@ def json_set(
...
+            The JSON path identifying the data that you want to obtain from the input.
+
+    Returns:
+        bigframes.series.Series: A new Series with the JSON or JSON-formatted STRING.
+    """
+    return series._apply_unary_op(ops.JSONExtract(json_path=json_path))
+
+
+# Search functions defined from
+# https://cloud.google.com/bigquery/docs/reference/standard-sql/search_functions
```
[the remark this answers is not in the document]
Comment (chelsea-lin) [reply bigframes/bigquery/__init__.py:243]: Yes, added it for `vector_search` to align with other methods.

File: tests/system/small/bigquery/test_json.py (hunk 1 above)
Comment (chelsea-lin) [reply tests/system/small/bigquery/test_json.py:133]: Thanks for looking the details. While the `dbjson` extension dtype is missing (working on [PR](https://github.com/googleapis/python-db-dtypes-pandas/pull/284)), we calls `_get_series_from_json` to return a JSON column from a BigQuery DataFrames. So the returned Series has a "data" name. That's why we have to ignore "check_names" in this test.

File: bigframes/operations/__init__.py (hunk 2 above)
Comment (chelsea-lin) [reply bigframes/operations/__init__.py:611]: `UnaryOp` indicates that we only have one input. Similarly, `BinaryOp` can have two inputs. More examples can be found in this file.

File: tests/system/small/bigquery/test_json.py (hunk 1 above)
Comment (shobsi) [reply tests/system/small/bigquery/test_json.py:133]: We are calling `_get_series_from_json` in both the tests, but using check_names=False only in this one, wondering why the difference

Comment (shobsi) [reply tests/system/small/bigquery/test_json.py:133]: Never mind, its because of the difference in `s = ...`

**Review by shobsi — APPROVED** [review]

**Pull request merged**